device/iothub_client_python/src/iothub_client_python.cpp

```diff
@@ -1225,11 +1225,10 @@ InboundDeviceMethodCallback(
     ScopedGILAcquire acquire;
     try
     {
-        boost::python::object user_response_obj = deviceMethodCallback(method_name_std_string, payload_std_string, method_id, userContext);
+        boost::python::object user_response_obj = deviceMethodCallback(method_name_std_string, payload_std_string, (unsigned long long)method_id, userContext);
 
-        DeviceMethodReturnValue user_response_value = boost::python::extract<DeviceMethodReturnValue>(user_response_obj)();
 
-        retVal = user_response_value.status;
+        retVal = boost::python::extract<int>(user_response_obj)();
```

Comment:
>  retVal = boost::python::extract<int>(user_response_obj)(); [](start = 6, length = 60)

Use the DeviceMethodReturnValue instead as in the DeviceMethodCallback. The code what you replaced should work.

Reply:
this method only expects the callback to return an integer and the response itself will be sent with IoTHubClient.device_method_response()